src/odemis/gui/comp/overlay/rectangle.py

```diff
@@ -88,18 +138,83 @@ def restore_state(self, state: RectangleState):
         self._points = self.get_physical_sel()
         self.points.value = self._points
 
-    def is_point_in_shape(self, point):
+    def is_point_in_shape(self, v_point):
         if self._points:
-            return point_in_polygon(point, self._points)
+            hover, _ = self.get_hover(v_point)
+            return bool(hover)
```

Comment:
This converts HOVER_* to a bool by relying on the fact that HOVER_NONE == 0. I think it'd be a lot more explicit and safe (for the long term) to write `return hover != HOVER_NONE`.